src/client/datascience/jupyter/kernels/cellExecution.ts

```diff
@@ -632,25 +632,27 @@ export class CellExecution {
         // eslint-disable-next-line complexity
         await chainWithPendingUpdates(this.editor.document, (edit) => {
             traceCellMessage(this.cell, 'Update streamed output');
-            let exitingCellOutput = this.cell.outputs;
+            let exitingCellOutputs = this.cell.outputs;
             // Clear output if waiting for a clear
             if (clearState.value) {
-                exitingCellOutput = [];
+                exitingCellOutputs = [];
                 clearState.update(false);
             }
 
-            // Ensure we append to previous output, only if the streams as the same.
-            // Possible we have stderr first, then later we get output from stdout.
-            // Basically have one output for stderr & a separate output for stdout.
-            // If we output stderr first, then stdout & then stderr, we should append the new stderr to the previous stderr output.
-            // Might already have a stream message. If so, just add on to it.
-            const existingItemToBeReplaced = exitingCellOutput.find(
-                (item) => item && isStreamOutput(item, msg.content.name)
-            );
-            // Get the jupyter output from the vs code output (so we can concatenate the text ourselves).
-            const outputs = existingItemToBeReplaced ? createIOutputFromCellOutputs([existingItemToBeReplaced]) : [];
-            if (existingItemToBeReplaced && outputs.length === 1 && nbformat.isStream(outputs[0])) {
-                let existingOutputText: string = concatMultilineString((outputs[0] as nbformat.IStream).text);
+            // Ensure we append to previous output, only if the streams as the same &
+            // If the last output is the desired stream type.
+            const lastOutput =
+                exitingCellOutputs.length > 0 ? exitingCellOutputs[exitingCellOutputs.length - 1] : undefined;
+            const existingOutputToAppendTo =
```

Comment:
Bug fixes, ensure each stream goes into a separate output.
Concatenate only if the last output is the same as the current output stream.

Reply:
Ie. do what happens in Jupyter